### PR TITLE
[FEATURE] Provide ability for DataAssistant to return its effective underlying BaseRuleBasedProfiler configuration

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
By returning `JSON` configuration of its underlying Rule-Based Profiler by a `DataAssistant`, it becomes possible to introspect this `JSON` structure and determine `runtime-environment` overrides programmatically.

Usage Example: 
```
context.assistants.onboarding.get_profiler_config()
```

By default , `JSON`-formatted structure is returned.  Other available formats are as per `ConfigOutputModes` in `great_expectations/core/config_peer.py`.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-1016
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/data_assistant/data_assistant_runner.py [hunk 1]
```diff
@@ -102,41 +104,95 @@ def __init__(
...
         batch_request: Optional[Union[BatchRequestBase, dict]] = None,
         **kwargs: dict,
     ) -> DataAssistantResult:
         """
-        variables: attribute name/value pairs, commonly-used in Builder objects, to modify using "runtime_environment"
-        rules: name/(configuration-dictionary) to modify using "runtime_environment"
-        kwargs: additional/override directives supplied at runtime
+        batch_request: Explicit batch_request used to supply data at runtime
+        kwargs: additional/override directives supplied at runtime (using "runtime_environment")
             "kwargs" directives structure:
```
Comment (Shinnnyshinshin) [inline great_expectations/rule_based_profiler/data_assistant/data_assistant_runner.py:161]: Very helpful to include this. Thank you Alex

File: great_expectations/rule_based_profiler/helpers/runtime_environment.py [hunk 2]
```diff
@@ -22,7 +22,13 @@ class RuntimeEnvironmentColumnDomainTypeDirectivesKeys(Enum):  # isort:skip
     EXCLUDE_COLUMN_NAME_SUFFIXES = "exclude_column_name_suffixes"
     INCLUDE_SEMANTIC_TYPES = "include_semantic_types"
     EXCLUDE_SEMANTIC_TYPES = "exclude_semantic_types"
+    ALLOWED_SEMANTIC_TYPES_PASSTHROUGH = "allowed_semantic_types_passthrough"
     CARDINALITY_LIMIT_MODE = "cardinality_limit_mode"
+    MAX_UNIQUE_VALUES = "max_unique_values"
```
Comment (Shinnnyshinshin) [inline great_expectations/rule_based_profiler/helpers/runtime_environment.py:27]: Not sure if I missed this, but I was wondering if these values were actually used in this PR.

**Review by Shinnnyshinshin — APPROVED** [review]
Approved, with one question :)

File: great_expectations/rule_based_profiler/helpers/runtime_environment.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/helpers/runtime_environment.py:27]: @Shinnnyshinshin this is for `runtime_environment` (the `kwargs` into `DataAssistant.run()`.  Thanks!

**Pull request merged**